### DISCUSSION TITLE
creating QR code from HL7-ORU

### DISCUSSION THREAD

**Discussion by Riven974** [nextgenhealthcare/connect#5833] [Q&A]
Hi all
do you know if it is possible to create a pdf with a QR code from HL7-ORU segments ? 
The idea will be to extract for exemple lastname and firstname from the HL7 segments, and do a QR code with them.
Thanks in advance for your help.
Véronique

**Comment by pacmano1** [comment]
Yes,it's possible. One example at https://stackoverflow.com/questions/37994970/using-mirth-connect-javascript-to-output-a-barcode-font-to-pdf, what have you tried?

**Reply by Riven974** [reply]
I haven't tryed yet. But I will try following the example you gave me. Thx !

**Comment by tonygermano** [comment]
I haven't tried it yet, but I just found this library, and it looks promising https://github.com/kenglxn/QRGen

It appears that it can even spit out the QR code as an svg, which I think can be added directly to the new document writer.

**Reply by Riven974** [reply]
Looks interesting ! I have heard about ZXING but never tried. Thanks, I will look also this solution.

**Comment by pacmano1** [comment]
@Riven974 , you sort this out?   If yes, please mark one of the responses as an answer.  It would also help the community to post your solution/code.
